ext/js/language/text-scanner.js

```diff
@@ -87,6 +87,9 @@ class TextScanner extends EventDispatcher {
         this._pointerIdTypeMap = new Map();
 
         this._canClearSelection = true;
+
+        this._yomichanIsChangingTextSelectionNow = false;
```

Comment:
Simplify names; something like `_ignoreSelectionChange` and `_hasManualUserSelection`

Reply:
First part:
* My rationale was that `_yomichan` is somewhat redundant since this class belongs to Yomichan, and I've tried to avoid redundant names on things (though there are [some places](https://github.com/FooSoft/yomichan/blob/4e7762edacb6cfdb49207bbf43102a04706c78a1/ext/js/pages/settings/anki-controller.js#L26-L48) I've not addressed).
* Similarly, the `Now` part is redundant since when you reference the value, it will always be "now".
* `_ignoreSelectionChange` may not be the best name, you're right, `_isChangingTextSelection` would probably have been a better suggestion, but my rationale is the same.
* Ultimately this is a semantics thing where your naming convention is different from how I usually do it, but probably not worth me getting hung up on.

Second part:
> Also, this inverts the logic. If a variable is only ever used with negation, it should not exist, IMO. 

Fair enough, as this would incur an additional negation unary on line 829. Aside from that, my rationale was that I have generally avoided storing negation variables, such as having "not" in the name, or preferring "enabled" rather than "disabled", for reasons of consistency. There is no difference in bytecode for either case.

Reply:
I used `yomichan` to signify that the add-on is changing the selection, as apposed to the user, or something else. I like it better than `addon` or `we`. With `selectionIsBeingChangedNow` you can't be sure, and `isChangingTextSelection` strongly suggests that we are *not* the actor, as stuff like `isConnected()` or `isBlankOrNull()` are often used to check the state of a thing that you do not directly set. Or at least I would find that confusing.

In a similar fashion, `yomichanIsChangingTextSelection` without `Now` could be read as a setting, as in the option to change selection is on. I think suffixes such as `Now` or `Flag` (I use this to signify that the thing is reset by the reader) are somewhat useful. (Also you would often use phrases ending with -ing for stuff like `with process_running() as pid: ...`, however, this can be seen as a counterexample maybe?)

It's not that I am a fan of long variable names, but in some cases I like them. Here, they don't hurt so to say. The statements all fit one line, and they only appear in a few places. If these were used extensively close together, it would make code blocks harder to read, but far apart it only helps, I think.

> I have generally avoided storing negation variables

I also avoid things like that! I guess this is a rare exception.